apis/core/config/crd/bases/core.tanzu.vmware.com_readinessproviders.yaml

```diff
@@ -0,0 +1,87 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.9.2
+  creationTimestamp: null
+  name: readinessproviders.core.tanzu.vmware.com
+spec:
+  group: core.tanzu.vmware.com
+  names:
+    kind: ReadinessProvider
+    listKind: ReadinessProviderList
+    plural: readinessproviders
+    singular: readinessprovider
+  scope: Cluster
+  versions:
+  - name: v1alpha2
+    schema:
+      openAPIV3Schema:
+        description: ReadinessProvider is the Schema for the readinessproviders API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: ReadinessProviderSpec defines the desired state of ReadinessProvider
+            properties:
+              checkName:
+                type: string
+              conditions:
+                items:
+                  properties:
+                    name:
+                      type: string
+                    resourceExistence:
+                      properties:
+                        group:
+                          type: string
+                        kind:
+                          type: string
+                        name:
+                          type: string
+                        namespace:
+                          type: string
+                        version:
+                          type: string
+                      required:
+                      - group
+                      - kind
+                      - name
+                      - namespace
```

Comment:
Should namespace be a required field ? How do we handle cluster-level resources ?

Reply:
Changed to optional. The controller can be coded such that entities with empty or nil namespace are treated as cluster scoped resources.

Reply:
This is a good example of using `*string` since we need to differentiate between:

1. `nil` namespace: meaning no namespace is set (cluster scoped resource).
2. Empty namespace: meaning namespace is set to `""`. This is not valid and client-go will return an error. We should not equate `nil` with `""` as it can be confusing.
3. Non-empty namespace for namespace-scoped resources.

Reply:
And this should be described properly in the doc string for this field.